source/grid/manifold.cc

```diff
@@ -101,23 +104,9 @@ get_new_point (const std::vector<Point<spacedim> > &surrounding_points,
   // First sort points in the order of their weights. This is done to
   // produce unique points even if get_intermediate_points is not
   // associative (as for the SphericalManifold).
-  unsigned int permutation_short[30];
-  std::vector<unsigned int> permutation_long;
-  unsigned int *permutation;
-  if (n_points > 30)
-    {
-      permutation_long.resize(n_points);
-      permutation = &permutation_long[0];
-    }
-  else
-    permutation = &permutation_short[0];
-
-  for (unsigned int i=0; i<n_points; ++i)
-    permutation[i] = i;
-
-  std::sort(permutation,
-            permutation + n_points,
-            CompareWeights(weights));
+  boost::container::small_vector<unsigned int, 100> permutation(n_points);
+  std::iota(permutation.begin(), permutation.end(), 0);
```

Comment:
It makes no difference here, but in general I've found that one needs to be careful with the last argument of functions such as `std::iota` and `std::accumulate` because all arithmetic is done in that data type when one typically would much rather prefer that argument to be converted to the `value_type` of the container one works on. So I might have written `0u` or `0U` in this place just to make the intent clear. But as I said, it makes no difference here.

Reply:
This is easy enough to fix that I went through all of our usages of `std::iota` and `std::accumulate` and fixed the type of the default element. I will put up a patch soon.

Reply:
Great, thanks!